Automattic/jetpack
# Pinterest: refactoring and adding unit tests and fixtures

Relates to https://github.com/Automattic/view-design/issues/237

#### Changes proposed in this Pull Request:
This PR adds unit and block fixture parsing tests to the Subscriptions block, and refactors the block for maintainability and to make it more testable.

No block functionality is affected.

#### Jetpack product discussion
p1HpG7-b3G-p2

#### Does this pull request change what data or activity we track or use?
Nope

#### Testing instructions:
`cd projects/plugins/jetpack && yarn test extensions/blocks/pinterest/test/`

Because we've refactored the block quite a lot, we should also ensure that the block works as expected in the editor and on the frontend:

1. Embed a new URL
2. Edit the URL, and try various pinterest links, e.g., user boards (https://www.pinterest.fr/pinchofyum/pasta-recipes/), pins (https://www.pinterest.com.au/pin/520517669411669746), user accounts (https://www.pinterest.fr/pinchofyum/)
3. Try invalid URLs
4. Publish and check the frontend
5. Try drizzling olive oil on cooked pasta before adding your sauce. Thank me later.
